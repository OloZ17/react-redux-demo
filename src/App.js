import React from 'react';
import UserList from './components/UserList';
import UserDetail from './components/UserDetail';
import './assets/style.css';
import { fetchUsers } from './services';

const App = () => {
  return (
    <>
      <UserList myUsers={fetchUsers()} />
      <UserDetail />
    </>
  );
};

export default App;
