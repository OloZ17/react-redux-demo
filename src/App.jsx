import React, { Component, Fragment } from "react";
import UserList from "./components/UserList";
import UserDetail from "./components/UserDetail";
import "./assets/style.css";
import { fetchUsers } from "./services";

export default class App extends Component {
  render() {
    return (
      <Fragment>
        <UserList myUsers={fetchUsers()} />
        <UserDetail />
      </Fragment>
    );
  }
}
