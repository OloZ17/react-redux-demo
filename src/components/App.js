import React, { Component, Fragment } from "react";
import UserList from "../containers/user_list";
import UserDetail from "../containers/user_detail";
import "../assets/style.css";
import { fetchUsers } from "../services";

export default class App extends Component {
  render() {
    // return <div>{UserList}</div>;
    //return <div>toto</div>;
    return (
      <Fragment>
        <UserList myUsers={fetchUsers()} />
        <UserDetail />
      </Fragment>
    );
  }
}
