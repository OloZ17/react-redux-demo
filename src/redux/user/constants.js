export const SELECT_USER = "USER/SELECT_USER";
