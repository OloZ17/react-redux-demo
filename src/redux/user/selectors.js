export const getActiveUser = state => state.activeUser || null;
