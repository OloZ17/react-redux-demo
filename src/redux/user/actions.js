import { SELECT_USER, RESET_USER } from './constants';

export const selectUser = (payload) => ({
  type: SELECT_USER,
  payload,
});

export const resetUser = () => ({
  type: RESET_USER,
});
