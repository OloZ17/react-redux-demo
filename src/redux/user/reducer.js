import { SELECT_USER } from "./constants";

const initialState = {
  id: 0,
  name: ""
};

export const user = (state = initialState, action) => {
  // console.log(state);
  switch (action.type) {
    case SELECT_USER:
      return action.payload;
    default:
      return state;
  }
};
