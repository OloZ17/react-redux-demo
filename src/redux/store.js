import { combineReducers, createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { user } from './user';

const rootReducer = combineReducers({
  activeUser: user,
});

// Bonus
const lastState = sessionStorage.getItem('redux-store')
  ? JSON.parse(sessionStorage.getItem('redux-store'))
  : {};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  rootReducer,
  lastState,
  composeEnhancers(applyMiddleware(thunk))
);

// Bonus
store.subscribe(() => {
  sessionStorage.setItem('redux-store', JSON.stringify(store.getState()));
});

export default store;
