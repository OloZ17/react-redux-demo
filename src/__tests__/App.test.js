import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render, cleanup } from '@testing-library/react';
import App from '../App';
import { user } from '../redux/user';

const renderWithRedux = (
  component,
  { initialState, store = createStore(user, initialState) } = {}
) => {
  return {
    ...render(<Provider store={store}>{component}</Provider>),
    store,
  };
};

afterEach(cleanup);

describe('Testing App() Component', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderWithRedux(<App />, {
      initialState: {
        activeUser: { id: 0, name: '' },
      },
    });

    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
