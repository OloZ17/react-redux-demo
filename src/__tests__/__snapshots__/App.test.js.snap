// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <ul
      class="col-md-4"
    >
      <li
        class="list-group-item"
      >
        Sarah Croche
      </li>
      <li
        class="list-group-item"
      >
        Eléonore  Labanquise
      </li>
      <li
        class="list-group-item"
      >
        Emma Karena
      </li>
      <li
        class="list-group-item"
      >
        Jean Bonnot
      </li>
    </ul>
  </div>
  <div>
    Sélectionner un utilisateur 
  </div>
</DocumentFragment>
`;
